pkg/horizon/secret.go

```diff
@@ -0,0 +1,33 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package horizon
+
+import (
+	"encoding/base64"
+
+	"math/rand"
+	"time"
+)
+
+// GenerateSecret -
+func GenerateSecret() string {
```

Comment:
it might make sense to add this to lib-common and pass in the pwd length? a while back we were talking about create service passwords if none were passed in via a secret which match the service selector. so this could be used there as well I think.

Reply:
+1. Let me work on the implementation in lib-common.

Reply:
I've pushed https://github.com/openstack-k8s-operators/lib-common/pull/172 .

Reply:
As is suggested in the above PR I'm trying the rand.String from apimachinery. Let me hold this PR until I test this out.